pkg/repository/backend_postgres_migrations/010_add_deployment_deleted_at_constraint.go

```diff
@@ -0,0 +1,30 @@
+package backend_postgres_migrations
+
+import (
+	"database/sql"
+
+	"github.com/pressly/goose/v3"
+)
+
+func init() {
+	goose.AddMigration(upAddDeploymentDeletedAtUniqueConstraint, dropDeploymentDeletedAtUniqueConstraint)
+}
+
+func upAddDeploymentDeletedAtUniqueConstraint(tx *sql.Tx) error {
+	_, err := tx.Exec(`ALTER TABLE deployment DROP CONSTRAINT deployment_name_version_workspace_id_stub_type_key RESTRICT`)
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(`ALTER TABLE deployment ADD CONSTRAINT unique_deploy UNIQUE (name, version, workspace_id, stub_type, deleted_at)`)
+	if err != nil {
```

Comment:
Probably don't need this if statement.

Reply:
actually i removed this. Talked with luke, its probably better to keep the old constraint but new deployments are created with a bumped version on the latest (even if latest was deleted)